### ISSUE TITLE
Lua Package Manager

### ISSUE THREAD

**Issue by rtsisyk** [tarantool/tarantool#160]
## Overview

It would be nice to have a tool to manage Lua modules.
I proposed to integrate [LuaRocks](http://gtihub.com/keplerproject/luarocks) to Tarantool.

Reasons:
- Organize distribution of modules supported by Tarantool Team (e.g. `tarantool-queue` or `expired`) and/or written by our customers
- Stimulate to write new opensource modules
- Allow our users to use modules made by Lua community
- Simplify development process (any required module can be locally installed by a developer)

Please note that `luarocks` is not intended to replace `dpkg` or `yum`. It is more like cpan, pip, gems and needed easy modules management during development. In my opinion all production system should use .deb or .rpms packages to install modules. We will make a script to automatically create dpkg/rpm rules/specs  from .rockspec (luarocks spec).
## Example

Query the repository for needed module:

```
$ taranrocks search queue

Search results:
===============


Rockspecs and source rocks:
---------------------------

tarantool-queue
   scm-1 (rockspec) - https://raw.github.com/rtsisyk/taranrocks-repo/master/
```

Install the module to the local tree (no root access is needed!):

```
$ taranrocks install tarantoool-queue --local 
Installing https://raw.github.com/rtsisyk/taranrocks-repo/master/tarantool-queue-scm-1.rockspec...
Using https://raw.github.com/rtsisyk/taranrocks-repo/master/tarantool-queue-scm-1.rockspec... switching to 'build' mode
Cloning into 'queue'...
Applying patch modularize...
successfully patched /tmp/luarocks_tarantool-queue-scm-1-2191/queue/init.lua
Updating manifest for /home/roman/.tarantool//share/tarantool/rocks

tarantool-queue scm-1 is now built and installed in /home/roman/.tarantool/ (license: Artistic)
```

Add to your init.lua:

```
-- Enable queue module
queue = require('tarantool.queue')
```

Restart Tarantool and connect to the admin console:

```
$ tarantool

---
- Tarantool 1.6.0-332-g8351f97
- 'Uptime: 2'
...

localhost> queue

---
- urgent: 'function: 0x4002c0f8'
  bury: 'function: 0x4003a500'
  put: 'function: 0x4002c0b0'
  -- // --
...
```

A queue server is up and running!
### Server
- Update package.path and package.cpath to include taranrocks directory
### Packaging
- Add https://github.com/keplerproject/luarocks (upstream) as a submodule to `third_party` or `src/lua` (TBD)
- Import https://github.com/rtsisyk/taranrocks/blob/master/bin/taranscript to emulate `#!/usr/bin/env tarantool_box --` behaviour on 1.5.x (and 1.6.x until #138 will be implemented)
- Import https://github.com/rtsisyk/taranrocks/blob/master/bin/taranrocks to `extra/taranrocks
- Update cmake rules to generate `luarocks/site_config.lua` as it is done by `./configure` script
- Install taranrocks to `${PREFIX}/bin/` on `make install`
- Install luarocks/src/luarocks to `${PREFIX}/share/tarantool/lua/`
- Install config.lua (luarocks configuration) to ${SYSCONFDIR}/etc/tarantool/rocks/config.lua
- (Optional, a separate ticket) Make a tool to automatically generate debian/rules and rpm/spec from .rockspec and put all modules from tarantool/rocks-main to repositories on tarantool.org

Source modules (*.lua) will be installed by taranrocks to `${PREFIX}/share/tarantool/lua` or `~./tarantool/share/tarantool/lua`

Binary modules (*.so) will be installed by taranrocks to `${PREFIX}/lib/tarantool/lua` or `~./tarantool/lib/tarantool/lua`
### GitHub
- Make two repositories
  - `tarantool/rocks-main` for modules that supported by Tarantool Team (e.g. tarantool-queue)
  - `tarantool/rocks-contrib` for third_party modules  (see https://github.com/rtsisyk/taranrocks-repo and https://github.com/keplerproject/rocks-scm)
    raw.github.com will be used as 
- Remove `rtsisyk/taranrocks` (content will be incorporated to `tarantool/tarantool`), and `rtsisyk/taranrocks-repo` (will be moved to `tarantool/rocks-contrib`)
- Add a script to generate a nice html/wiki page with packages information

This proposal is not blocked by #138 or #139
